File: Data Structures & Algorithms/remove-element/submission-3.java
class Solution {
    public int removeElement(int[] nums, int val) {
        if(nums.length == 0){
            return 0;
        }
        
        Integer left = 0;
        Integer right = nums.length;
        
        while (right > left){
            if (nums[left] == val){
                nums[left] = nums[--right];
            }
            else{
                left += 1;
            }
        }

        return right;
    }
}

File: Data Structures & Algorithms/permutation-string/submission-3.py
class Solution:
    def checkInclusion(self, s1: str, s2: str) -> bool:
        s1Freq = Counter(s1)
        s2Freq = Counter()

        left = 0;

        for right in range(len(s2)):
            s2Freq[s2[right]] += 1

            if (right - left + 1 > len(s1)):
                
                ## move left 
                s2Freq[s2[left]] -= 1
                if (s2Freq[s2[left]] == 0):
                    del s2Freq[s2[left]]
                left += 1 
                
            
            if s1Freq == s2Freq:
                return True


        return False